src/services/userService.ts

```diff
@@ -0,0 +1,57 @@
+import { sign } from "jsonwebtoken";
+import { compare, hash } from "bcryptjs";
+import IUser from "interfaces/userInterface";
+import { User } from "models/userModel";
+import { appConfig } from "config/app-config";
+
+export const createUser = async (obj: IUser) => {
+  obj.password = await hash(obj.password, 10);
+  const user = await User.create({ ...obj });
+  return user;
+};
+
+export const addUser = async (obj: IUser) => {
+  const user: IUser[] = await User.find({ name: obj.name });
```

Comment:
use findOne() instead of find(), so that no need to use it like user[0].password or user[0].name